### ISSUE TITLE
Linux x86-64 binaries are linked with libclang (Origin: bugzilla #792761)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6317]
status NEW severity *normal* in component *build* for *---*
Reported in version *1.8.14* on platform *Other*
Assigned to: Dimitri van Heesch

On 2018-01-21 20:04:46 +0000, Daniel wrote:

    The Linux binaries from
    http://ftp.stack.nl/pub/users/dimitri/doxygen-1.8.14.linux.bin.tar.gz are
    linked with libclang:
    
    ldd doxygen
    ./doxygen: /lib/x86_64-linux-gnu/libtinfo.so.5: no version information
    available (required by ./doxygen)
    	linux-vdso.so.1 =>  (0x00007ffc1de99000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0
    (0x00007f12ed2cd000)
    	libclang.so.6 => not found
    	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f12ed0a4000)
    	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f12ece8b000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    (0x00007f12ecb78000)
    	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f12ec872000)
    	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f12ec65b000)
    	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f12ec292000)
    	/lib64/ld-linux-x86-64.so.2 (0x00007f12ed4eb000)

**Comment by datalogics-kam** [comment]
Thinks for continuing to maintain Doxygen!

In [this comment](https://github.com/doxygen/doxygen/issues/6393#issuecomment-407696011) you mention that you statically linked libclang on Linux. This would help our project out immensely as we use Doxygen via Conan using the [inexorgame/conan-doxygen](https://github.com/inexorgame/conan-doxygen) project.

I noticed the website moved around, and I can't find the statically-linked Linux version any more. Is it around somewhere? Could a newer release of Doxygen support static libclang?

**Comment by albert-github** [comment]
Did you have a look at: http://www.doxygen.nl/download.html ?
Is the file you need: http://doxygen.nl/files/doxygen-1.8.14.linux.bin.tar.gz ?

**Comment by datalogics-kam** [comment]
If I download `doxygen-1.8.14.linux.bin.tar.gz`, it doesn't have the necessary `libclang.so`, where the Mac and Windows versions include a `.dylib` or `.dll`.

```
kam@bison-centos6 /tmp/doxygen-1.8.14 ❯❯❯ bin/doxygen
bin/doxygen: error while loading shared libraries: libclang.so.6: cannot open shared object file: No such file or directory
```

Since I'm packaging this for Conan, it would be helpful if all needed libraries were in the tarball. I don't want to have a global system requirement. Also, up-to-date LLVM toolchains are hard to come by on CentOS.

My current angle is to build from scratch without Clang support, but it'd be great to have a prepackaged binary that just works out of the box.

**Comment by Croydon** [comment]
The issue is solved. I would wish however that @doxygen is shining a light on how the official Linux binaries are build. Is the static linking of libclang now fully integrated in the CMake code of doxygen? Or was this done manually, if so, how -exactly-?

**Comment by doxygen** [comment]
Doxygen was built using:

```
cmake -DCMAKE_BUILD_TYPE=Release \
      "-DCMAKE_FIND_LIBRARY_SUFFIXES=.a" \
      "-ldl;-lz;-ltinfo;-lpthread" \
      -Duse_libclang=YES \
      path_to_doxygen_root_source_dir
```

The prerequisite is a statically build version of LLVM/clang.

This can be done by downloading the sources for LLVM and clang and then build them with the following options:

```
cmake -DLIBCLANG_BUILD_STATIC=ON \
      -DBUILD_SHARED_LIBS=OFF \
      -DLLVM_ENABLE_PIC=OFF \
      -DLLVM_BUILD_LLVM_DYLIB=OFF \
      -DLLVM_BUILD_LLVM_C_DYLIB=OFF \
       path_to_llvm_root_source_dir
```

**Comment by PiJ82** [comment]
@doxygen Does it make sense to add those information on the [installation section of the doxygen manual](http://www.doxygen.nl/manual/install.html)?

**Comment by doxygen** [comment]
I've added it (see this commit https://github.com/doxygen/doxygen/commit/05ac75efb2d4467214acc350ada97da86504e333)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
